parachain-upgrade/src/lib.rs

```diff
@@ -103,52 +106,60 @@ decl_module! {
 		/// As a side effect, this function upgrades the current validation function
 		/// if the appropriate time has come.
 		#[weight = (0, DispatchClass::Mandatory)]
-		fn set_validation_function_parameters(origin, vfp: ValidationFunctionParams) {
+		fn set_validation_data(origin, vfp: ValidationData) {
 			ensure_none(origin)?;
-			assert!(!DidUpdateVFPs::exists(), "VFPs must be updated only once in the block");
+			assert!(!DidUpdateValidationData::exists(), "ValidationData must be updated only once in a block");
 
 			// initialization logic: we know that this runs exactly once every block,
 			// which means we can put the initialization logic here to remove the
 			// sequencing problem.
 			if let Some((apply_block, validation_function)) = PendingValidationFunction::get() {
-				if vfp.relay_chain_height >= apply_block {
+				if vfp.persisted.block_number >= apply_block {
 					PendingValidationFunction::kill();
 					Self::put_parachain_code(&validation_function);
-					Self::deposit_event(Event::ValidationFunctionApplied(vfp.relay_chain_height));
+					Self::deposit_event(Event::ValidationFunctionApplied(vfp.persisted.block_number));
 				}
 			}
 
-			storage::unhashed::put(VALIDATION_FUNCTION_PARAMS, &vfp);
-			DidUpdateVFPs::put(true);
-			<T::OnValidationFunctionParams as OnValidationFunctionParams>::on_validation_function_params(vfp);
+			storage::unhashed::put(VALIDATION_DATA, &vfp);
```

Comment:
Can you explain the new relationship between `ValidationData` and `ValidationParams` and for which situations each is relevant?

Reply:
`ValidationData` is passed using an inherent into the runtime, while `ValidationParams` are passed when validating the parachain.

We use the `ValidationParams` to validate the `ValidationData` while validating the PovBlock.